src/apis/routine.api.ts

```diff
@@ -0,0 +1,73 @@
+import { Router } from 'express';
+import { body } from 'express-validator';
+
+import updateLectureDetails from '../controllers/routine/update-lecture.controller';
+import deleteLecture from '../controllers/routine/delete-lecture';
+import createNewLecture from '../controllers/routine/new-lecture.controller';
+import hasType from '../middlewares/has-type';
+import isAuth from '../middlewares/is-auth';
+import validationErrorHandler from '../middlewares/validation-error-handler';
+import getLecturesByGrade from '../controllers/routine/get-lectures-by-grade';
+
+const router = Router();
+
+/**
+ * @description - These routes are used for routines CRUD operations
+ * @auth - required
+ * @route /routine
+ * */
+
+router.post(
+  '/create_new_lecture',
+  [
+    body('name').notEmpty(),
+    body('subject').notEmpty(),
+    body('day').notEmpty(),
+    body('startTime').notEmpty(),
+    body('endTime').notEmpty(),
+  ],
+  validationErrorHandler,
+  isAuth,
+  hasType(['PRINCIPAL', 'SENIOR_TEACHER']),
+  createNewLecture,
+);
+
+router.delete(
+  '/delete_lecture',
+  [body('lectureId').notEmpty()],
+  validationErrorHandler,
+  isAuth,
+  hasType(['PRINCIPAL', 'SENIOR_TEACHER']),
+  deleteLecture,
+);
+
+router.post(
+  '/update_lecture_details',
+  isAuth,
+  hasType(['PRINCIPAL', 'SENIOR_TEACHER']),
+  [body('id').notEmpty()],
+  validationErrorHandler,
+  updateLectureDetails,
+);
+
+router.post(
```

Comment:
use get method for fetching details related apis
and take necessary parameter from query or param

Reply:
@Prodip-Kumar-Paul IG it doesn't matter. Will change fro next APIs